src/Core/Settings.h

```diff
@@ -468,6 +468,7 @@ class IColumn;
     M(Bool, output_format_enable_streaming, false, "Enable streaming in output formats that support it.", 0) \
     M(Bool, output_format_write_statistics, true, "Write statistics about read rows, bytes, time elapsed in suitable output formats.", 0) \
     M(Bool, allow_non_metadata_alters, true, "Allow to execute alters which affects not only tables metadata, but also data on disk", 0) \
+    M(Bool, with_global, false, "Enable WITH statements for UNION queries", 0) \
```

Comment:
`global_with` sounds better
`allow_global_with` or `enable_global_with` even better

Reply:
We don't have `enable` neither `allow` for every setting that enables or allows something...

Reply:
It is good to have a verb in boolean setting name. e.g. `output_format_write_statistics`